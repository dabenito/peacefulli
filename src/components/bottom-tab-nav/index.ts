export { Nav } from "./bottom-tab-nav";
