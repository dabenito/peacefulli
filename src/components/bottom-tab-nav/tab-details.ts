import { ComponentClass } from "react";
import { Home } from "../home";
import { Support } from "../support";
import { Profile } from "../profile";
import { ListView } from "../list-view";

// TODO: This needs to be localized
enum TabName {
  SUPPORT = "SUPPORT",
  HOME = "HOME",
  LIST = "LIST",
  PROFILE = "PROFILE",
}

export type TabDetails = {
  name: string;
  component: ComponentClass<any, any>;
  icon: string;
};

export namespace TabInfo {
  export const SUPPORT: TabDetails = {
    name: TabName.SUPPORT,
    component: Support,
    icon: "hand-holding-heart",
  };
  export const HOME: TabDetails = {
    name: TabName.HOME,
    component: Home,
    icon: "home",
  };
  export const LIST: TabDetails = {
    name: TabName.LIST,
    component: ListView,
    icon: "list",
  };
  export const PROFILE: TabDetails = {
    name: TabName.PROFILE,
    component: Profile,
    icon: "user-circle",
  };
}
