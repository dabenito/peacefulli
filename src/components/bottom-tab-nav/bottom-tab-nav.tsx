import React, { PureComponent } from "react";
import AppLoading from "expo-app-loading";
import * as Font from "expo-font";
import Icon from "react-native-vector-icons/FontAwesome5";
import { NavigationContainer } from "@react-navigation/native";
import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
import { Color } from "../../constants/color-constants";
import { styles } from "./style";
import { TabDetails, TabInfo } from "./tab-details";
import { customFonts } from "../../constants/font-constants";

const Tab = createBottomTabNavigator();

export interface BottomTabNavProps {
  defaultTab: string;
}

export interface BottomTabNavState {
  fontLoaded: boolean;
}

export class BottomTabNav extends PureComponent<
  BottomTabNavProps,
  BottomTabNavState
> {
  constructor(props: BottomTabNavProps) {
    super(props);
    this.state = { fontLoaded: false };
  }

  create_tab(tabDetails: TabDetails): JSX.Element {
    return (
      <Tab.Screen
        name={tabDetails.name}
        component={tabDetails.component}
        options={{
          tabBarLabel: tabDetails.name,
          tabBarIcon: ({ focused }) => {
            const color = focused ? Color.pink : Color.purple;
            return (
              <Icon
                name={tabDetails.icon}
                size={styles.tabIcon.height}
                color={color}
              />
            );
          },
        }}
      />
    );
  }

  render() {
    if (!this.state.fontLoaded) {
      return (
        <AppLoading
          startAsync={async () => {
            await Font.loadAsync(customFonts);
          }}
          onFinish={() => this.setState({ fontLoaded: true })}
          onError={console.warn}
        />
      );
    }
    return (
      <NavigationContainer>
        <Tab.Navigator
          initialRouteName={this.props.defaultTab}
          tabBarOptions={{
            activeTintColor: Color.pink,
            inactiveTintColor: Color.purple,
            labelStyle: styles.text,
            style: styles.tabContainer,
          }}
        >
          {this.create_tab(TabInfo.LIST)}
          {this.create_tab(TabInfo.HOME)}
          {this.create_tab(TabInfo.SUPPORT)}
          {this.create_tab(TabInfo.PROFILE)}
        </Tab.Navigator>
      </NavigationContainer>
    );
  }
}
