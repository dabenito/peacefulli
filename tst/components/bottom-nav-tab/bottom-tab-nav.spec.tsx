import React from "react";
import { cleanup } from "@testing-library/react-native";
import { shallow, configure } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { BottomTabNav } from "../../../src/components/bottom-tab-nav/bottom-tab-nav";

jest.mock("react-native-vector-icons/FontAwesome5", () => "Icon");

configure({ adapter: new Adapter() });

describe("BottomTabNav Tests", () => {
  afterEach(cleanup);

  it("loading nav with the LIST default", async () => {
    const nav = shallow(<BottomTabNav defaultTab="LIST" />);
    nav.setState({ fontLoaded: true });
    expect(shallowToJson(nav)).toMatchSnapshot();
  });

  it("loading nav with the HOME default", async () => {
    const nav = shallow(<BottomTabNav defaultTab="HOME" />);
    nav.setState({ fontLoaded: true });
    expect(shallowToJson(nav)).toMatchSnapshot();
  });

  it("loading nav with the SUPPORT default", async () => {
    const nav = shallow(<BottomTabNav defaultTab="SUPPORT" />);
    nav.setState({ fontLoaded: true });
    expect(shallowToJson(nav)).toMatchSnapshot();
  });

  it("loading nav with the PROFILE default", async () => {
    const nav = shallow(<BottomTabNav defaultTab="PROFILE" />);
    nav.setState({ fontLoaded: true });
    expect(shallowToJson(nav)).toMatchSnapshot();
  });

  it("font not loaded", async () => {
    const nav = shallow(<BottomTabNav defaultTab="LIST" />);
    nav.setState({ fontLoaded: false });
    expect(shallowToJson(nav)).toMatchSnapshot();
  });
});
