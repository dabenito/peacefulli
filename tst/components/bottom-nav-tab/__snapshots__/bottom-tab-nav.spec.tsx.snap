// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTabNav Tests font not loaded 1`] = `
<AppLoading
  onError={[Function]}
  onFinish={[Function]}
  startAsync={
    Promise {
      "_U": 0,
      "_V": 0,
      "_W": null,
      "_X": null,
    }
  }
/>
`;

exports[`BottomTabNav Tests loading nav with the HOME default 1`] = `
<ForwardRef(NavigationContainer)>
  <BottomTabNavigator
    initialRouteName="HOME"
    tabBarOptions={
      Object {
        "activeTintColor": "#FF00F0",
        "inactiveTintColor": "#6900FF",
        "labelStyle": Object {
          "fontFamily": "pt_sans_bold",
          "fontWeight": "bold",
        },
        "style": Object {
          "paddingHorizontal": 30,
          "paddingTop": 5,
        },
      }
    }
  >
    <Screen
      component={[Function]}
      name="LIST"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "LIST",
        }
      }
    />
    <Screen
      component={[Function]}
      name="HOME"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "HOME",
        }
      }
    />
    <Screen
      component={[Function]}
      name="SUPPORT"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "SUPPORT",
        }
      }
    />
    <Screen
      component={[Function]}
      name="PROFILE"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "PROFILE",
        }
      }
    />
  </BottomTabNavigator>
</ForwardRef(NavigationContainer)>
`;

exports[`BottomTabNav Tests loading nav with the LIST default 1`] = `
<ForwardRef(NavigationContainer)>
  <BottomTabNavigator
    initialRouteName="LIST"
    tabBarOptions={
      Object {
        "activeTintColor": "#FF00F0",
        "inactiveTintColor": "#6900FF",
        "labelStyle": Object {
          "fontFamily": "pt_sans_bold",
          "fontWeight": "bold",
        },
        "style": Object {
          "paddingHorizontal": 30,
          "paddingTop": 5,
        },
      }
    }
  >
    <Screen
      component={[Function]}
      name="LIST"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "LIST",
        }
      }
    />
    <Screen
      component={[Function]}
      name="HOME"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "HOME",
        }
      }
    />
    <Screen
      component={[Function]}
      name="SUPPORT"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "SUPPORT",
        }
      }
    />
    <Screen
      component={[Function]}
      name="PROFILE"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "PROFILE",
        }
      }
    />
  </BottomTabNavigator>
</ForwardRef(NavigationContainer)>
`;

exports[`BottomTabNav Tests loading nav with the PROFILE default 1`] = `
<ForwardRef(NavigationContainer)>
  <BottomTabNavigator
    initialRouteName="PROFILE"
    tabBarOptions={
      Object {
        "activeTintColor": "#FF00F0",
        "inactiveTintColor": "#6900FF",
        "labelStyle": Object {
          "fontFamily": "pt_sans_bold",
          "fontWeight": "bold",
        },
        "style": Object {
          "paddingHorizontal": 30,
          "paddingTop": 5,
        },
      }
    }
  >
    <Screen
      component={[Function]}
      name="LIST"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "LIST",
        }
      }
    />
    <Screen
      component={[Function]}
      name="HOME"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "HOME",
        }
      }
    />
    <Screen
      component={[Function]}
      name="SUPPORT"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "SUPPORT",
        }
      }
    />
    <Screen
      component={[Function]}
      name="PROFILE"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "PROFILE",
        }
      }
    />
  </BottomTabNavigator>
</ForwardRef(NavigationContainer)>
`;

exports[`BottomTabNav Tests loading nav with the SUPPORT default 1`] = `
<ForwardRef(NavigationContainer)>
  <BottomTabNavigator
    initialRouteName="SUPPORT"
    tabBarOptions={
      Object {
        "activeTintColor": "#FF00F0",
        "inactiveTintColor": "#6900FF",
        "labelStyle": Object {
          "fontFamily": "pt_sans_bold",
          "fontWeight": "bold",
        },
        "style": Object {
          "paddingHorizontal": 30,
          "paddingTop": 5,
        },
      }
    }
  >
    <Screen
      component={[Function]}
      name="LIST"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "LIST",
        }
      }
    />
    <Screen
      component={[Function]}
      name="HOME"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "HOME",
        }
      }
    />
    <Screen
      component={[Function]}
      name="SUPPORT"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "SUPPORT",
        }
      }
    />
    <Screen
      component={[Function]}
      name="PROFILE"
      options={
        Object {
          "tabBarIcon": [Function],
          "tabBarLabel": "PROFILE",
        }
      }
    />
  </BottomTabNavigator>
</ForwardRef(NavigationContainer)>
`;
